### DISCUSSION TITLE
`formatter` -fix removes `fmt.Sprintf` but doesn't suffix assertion method with `f`

### DISCUSSION THREAD

**Discussion by mmorel-35** [Antonboom/testifylint#186] [Q&A] [answered]
With master version of testifylint, with (Example from https://github.com/kubernetes/kubernetes/pull/127649/files)

```go 
assert.Error(t, err, fmt.Sprintf("Profile %s should not be valid", test.profile))
```

The suggestion is

```go 
assert.Error(t, err, "Profile %s should not be valid", test.profile)
```

But I expect 

```go 
assert.Errorf(t, err, "Profile %s should not be valid", test.profile)
```

**Comment by Antonboom** [comment]
@mmorel-35 hi!

did you use `--formatter.require-f-funcs` flag?

(disabled by default to avoid a lot of changes in large codebases, read the checker docs)

**Comment by mmorel-35** [comment]
With `--formatter.require-f-funcs` it adds the `f` even if there are no argument to the message:

While with : 
```go
assert.Error(t, err, "Profile should not be valid")
```

I don't expect to have 
```go
assert.Errorf(t, err, "Profile should not be valid")
```
but with:
```go
assert.Error(t, err, "Profile %s should not be valid", test.profile)
```
I expect
```go
assert.Errorf(t, err, "Profile %s should not be valid", test.profile)
```

**Comment by Antonboom** [comment, marked as answer]
Your expectations are invalid.
Please, read the [checker doc](https://github.com/Antonboom/testifylint?tab=readme-ov-file#formatter) carefully, including historical reference.

By Go convention f-suffix is triggered by any message, without arguments too.
Look at `fmt.Sprint` and `fmt.Sprintf` signatures for example.
